sqlglot/dialects/hive.py

```diff
@@ -121,6 +122,33 @@ def _index_sql(self, expression):
     return f"{this} ON TABLE {table} {columns}"
 
 
+def _create_sql(self, expression):
+    """
+    In Hive and Spark, the PARTITIONED BY property acts as an extension of a table's schema. When the
+    PARTITIONED BY value is an array of column names, they are transformed into a schema. The corresponding
+    columns are removed from the create statement.
+    """
+    schema = isinstance(expression.this, exp.Schema) and expression.this
+    properties = expression.args.get("properties")
+    is_partitionable = expression.args.get("kind") in ("TABLE", "VIEW")
+
+    if schema and properties and is_partitionable:
+        for prop in properties.expressions:
+            value = prop.args.get("value")
+            if isinstance(prop, exp.PartitionedByProperty) and value and not isinstance(value, exp.Schema):
+
+                columns = {v.name.upper() for v in value.expressions}
+
+                partitions = [col for col in schema.expressions if col.name.upper() in columns]
+                schema.set(
```

Comment:
this is a mutation, we shouldn't do mutations in the generator.

do

schema = (isinstance ...).copy()

Reply:
I've moved this to dialects.py and re-wrote it so it doesnt perform any mutations.